## Share your thinking process here

1. What is the bioinformatics challenge your scripts solve?
My script solves the calculations of breadth of coverage and mean depth of coverage for targeted genomic sequences (Golden genes and unrequested regions), as well as the breadth of coverage and mean depth of coverage for the panel.
Additionally, another script generates a descriptive plot of the data

2. Why is important to solve this bioinformatics challenge (think like the scientist that requested the solution to the problem at hand)?

It is important to solve the bioinformatic challenge so it will provide the necesarry information to elucidate, whether the design of the panel is adecuate to obtain the the specific DNA region of interest. The coverage calculations will provide information about the specificity of the probes in the panel to identify the targeted region and inform about its reliability.

3. How do your scripts solve the bioinformatics challenge? (Explain briefly how your scripts work)

To obtain the breadth of coverage, my script sums only the bases in the genomic region of interest, that are covered by at least one read, and divides this number by the length of the genomic region times 100 to obtain the percentage.

In the case of mean depth of coverage, my script sums the mapped sequencing reads of every covered base, to be devided by the number of bases in the genomic region of interest that are covered by at least one sequencing read.

4. Show a tree description of the final file structure of your process directory (see ../README.md for an example)

````
Bioinformatician-001/		## The original cloned repository directory
├── process				## This is the directory where my code is.
│   ├── README.md			## A readme file explaining my process.
│   ├── plotter.R			## My Rscript to plot the requested figure.
│   └── coverage_calculator.sh		## My main bash script for calculating the requested coverage values.
├── README.md				## This README file, explaining the test
├── results-examples	    		## We provide examples of the requested outputs, your results should be as close to this as possible
│   ├── coverage_calculations.tsv	## An example of the requested results, generated by Winter Genomics team
|   |---coverage_calculations.csv       ## Results generated by klg
|   |---Results_report.pdf		## Final report that answers the original question.
│   └── final_report.md			## An example of the report that answers the original questions posed by the PROBLEM 
DESCRIPTION
└── test-materials			## In this directory we provide the necessary data to generate the requested results
    └── test-data
        └── raw_coverages_by_base.tsv    ## A raw coverage file. Was generated with bedtools v2.25.0 using the -d option, using a public BAM file and a focused BED file for the loci of interest

````
